### ISSUE TITLE
plugins.py:98: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.

### ISSUE THREAD

**Issue by alexamici** [pydata/xarray#6514]
### What happened?

The plugin interface warns with `DeprecationWarning` in python 3.10.

### What did you expect to happen?

No warning to be issued.

### Minimal Complete Verifiable Example

```Python
Running `pytest` on any *Xarray* importing test.
```

### Relevant log output

```Python
   .../python3.10/site-packages/xarray/backends/plugins.py:98: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
    entrypoints = entry_points().get("xarray.backends", ())
```

### Anything else we need to know?

According to the *Compatibity note* in:

https://docs.python.org/3/library/importlib.metadata.html#entry-points

the new API is `entry_points(group='xarray.backends')`

### Environment

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 3.10.4 | packaged by conda-forge | (main, Mar 24 2022, 17:42:03) [Clang 12.0.1 ]
python-bits: 64
OS: Darwin
OS-release: 21.3.0
machine: arm64
processor: arm
byteorder: little
LC_ALL: None
LANG: None
LOCALE: (None, 'UTF-8')
libhdf5: None
libnetcdf: None

xarray: 2022.3.0
pandas: 1.4.2
numpy: 1.22.3
scipy: 1.8.0
netCDF4: None
pydap: None
h5netcdf: None
h5py: None
Nio: None
zarr: 2.11.3
cftime: None
nc_time_axis: None
PseudoNetCDF: None
rasterio: 1.2.10
cfgrib: None
iris: None
bottleneck: None
dask: 2022.04.1
distributed: 2022.4.1
matplotlib: 3.5.1
cartopy: None
seaborn: None
numbagg: None
fsspec: 2022.3.0
cupy: None
pint: None
sparse: None
setuptools: 62.1.0
pip: 22.0.4
conda: 4.12.0
pytest: 7.1.2
IPython: 8.2.0
sphinx: None
</details>

**Comment by bcbnz** [comment]
Just came to report this after seeing it in some tests of one of my projects. Based on a change I made for plugins in that project, I think that changing

https://github.com/pydata/xarray/blob/d479009d79374dc4a56c9f4346b1af38f5ac182c/xarray/backends/plugins.py#L96-L99

to

```python
    @functools.lru_cache(maxsize=1)
    def list_engines():
        try:
            entrypoints = entry_points(group="xarray.backends")  # Python >= 3.10
        except TypeError:
            entrypoints = entry_points().get("xarray.backends", ())
        return build_engines(entrypoints)
```

will take care of this. Not tested as I don't have access to an older Python environment right now, but I can make a pull request later if this would be a suitable workaround and nobody else does so first.

**Comment by dcherian** [comment]
@bcbnz please open a PR if you have the time.  

One suggestion would be to use an explicit version check using ` sys.version_info()` instead of catching `TypeError`

**Comment by bcbnz** [comment]
> The _Compatibility Note_ I linked above appears to point out the the `entry_points(group="xarray.backends")` is available since Python 3.6, so we not need the versione check at all.

There's some unfortunate version numbers there. That refers to version 3.6 of the third-party [importlib_metadata](https://pypi.org/project/importlib-metadata/) library, a later version of which was used for the ``importlib.metadata`` library included with Python 3.10. If you try to run `entry_points(group="xarray.backends")` on Python 3.9 it fails with a TypeError.
